hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/metrics/NameNodeMetrics.java

```diff
@@ -87,6 +81,19 @@
   MutableGaugeInt blockOpsQueued;
   @Metric("Number of blockReports and blockReceivedAndDeleted batch processed")
   MutableCounterLong blockOpsBatched;
+  @Metric("Number of snapshot GC runs")
+  MutableCounterLong numSnapshotGcRun;
+  @Metric("Number of empty snapshot GC runs")
+  MutableCounterLong numSnapshotGcRunEmpty;
```

Comment:
Any specific use case for numSnapshotGcRunEmpty metric ??

Reply:
numSnapshotGcRunEmpty would be useful BDR runs where we can track the number of non-GCed cycles.